rust/connlib/clients/shared/src/lib.rs

```diff
@@ -114,130 +98,25 @@ where
             }));
         }
 
-        Self::connect_inner(
-            &runtime,
-            tx,
-            api_url.try_into().map_err(|_| Error::UriError)?,
+        runtime.spawn(connect(
```

Comment:
Should we create a single-threaded runtime? There's no point now to having a multi-threaded one

Reply:
I wanted to keep the changes as minimal as possible but yes, we should make a single-threaded on in the future.